indigo_app/static/javascript/indigo/views/document_toc.js

```diff
@@ -6,266 +6,34 @@
-          toc = this.toc;
-
-      // first, find the closest element's ancestor that is a toc element
-      while (element) {
-        if (tradition.is_toc_element(element)) {
-          // now get the toc item for this element
-          for (var i = 0; i < toc.length; i++) {
-            if (toc[i].element === element) return toc[i];
-          }
-        }
 
-        element = element.parentElement;
+      this.compInstance = null;
+      if (this.registeredComp && this.targetMountElement) {
+        this.render();
       }
     },
-
     render: function() {
-      // recursively render the TOC
-      function renderItem(root, item) {
-        var li = document.createElement('li');
-        li.classList.add('toc-item');
-
-        // issues?
-        if (item.issues && item.issues.length) {
-          var icon = document.createElement('i');
-          icon.className = 'float-right issue-icon issue-' + item.issues_severity;
-          li.appendChild(icon);
-
-          $(icon).popover({
-            content: item.issues_description,
-            title: item.issues_title,
-            trigger: 'hover',
-            placement: 'bottom',
-            html: true,
-          });
-        }
-
-        var a = document.createElement('a');
-        a.setAttribute('href', '#');
-        a.setAttribute('data-index', item.index);
-        a.textContent = item.title;
-        if (item.selected) a.classList.add('active');
-        li.appendChild(a);
-
-        if (item.children) {
-          var kids = document.createElement('ol');
-          li.appendChild(kids);
-          item.children.forEach(function(x) { renderItem(kids, x); });
-        }
-
-        root.appendChild(li);
-      }
-
-      var root = document.createElement('ol');
-      this.roots.forEach(function(x) { renderItem(root, x); });
-      this.$el.empty().append(root);
+      var CompClass = window.globalVue.extend(this.registeredComp);
+      var compInstance = new CompClass({
+        propsData: { backBoneContext: this }
+      });
+      compInstance.$mount(this.targetMountElement);
+      this.compInstance = compInstance;
     },
 
-    // select the i-th item in the TOC
     selectItem: function(i, force) {
-      var index = this.selection.get('index');
-
-      i = Math.min(this.toc.length-1, i);
-
-      if (force || index !== i) {
-        // unmark the old one
-        if (index > -1 && index < this.toc.length) {
-          delete (this.toc[index].selected);
-        }
-
-        if (i > -1) {
-          this.toc[i].selected = true;
-        }
-
-        this.render();
-
-        // only do this after rendering
-        if (force) {
-          // ensure it forces a change
-          this.selection.clear({silent: true});
-        }
-        this.selection.set(i > -1 ? this.toc[i] : {});
+      if (this.compInstance) {
+        this.compInstance.selectItem(i, force);
       }
-    },
-
-    selectItemById: function(itemId) {
```

Comment:
You probably need to write a delegating function like you did for selectItem, because this is likely called from elsewhere in Indigo.

Reply:
done